internal/toolexec/aspect/resolve.go

```diff
@@ -19,79 +19,61 @@ import (
-// invoked to create a new archive; and `go list` is then used again to obtain the path of the
-// archives.
+// to locate the archive for `importPath` and its dependencies using `go list`. If that fails, it
+// will try to resolve it using `go get`.
 func resolvePackageFiles(importPath string) (map[string]string, error) {
 	// Apply quoting as appropriate to avoid shell interpretation issues...
 	toolexec := fmt.Sprintf("%q %q", os.Args[0], os.Args[1])
 
-	attemptedBuild := false // Whether we attempted a build already or not
 	// Retrieve parent Go command flags
 	args, err := prepareGoCommandArgs("list", "-toolexec", toolexec, "-json", "-deps", "-export", "--", importPath)
 	if err != nil {
 		return nil, fmt.Errorf("preparing go command %v: %w", args, err)
 	}
-	for {
-		cmd := exec.Command("go", args...)
-		var stdout bytes.Buffer
-		cmd.Stdout = &stdout
-		log.Tracef("Attempting to resolve %q using %q\n", importPath, cmd.Args)
-		if err = cmd.Run(); err != nil {
-			return nil, fmt.Errorf("running %q: %w", cmd.Args, err)
-		}
 
-		type listItem struct {
-			ImportPath string // The import path of the package
-			Export     string // The path to its archive, if any
-			BuildID    string // The build ID for the package
-			Standard   bool   // Whether this is from the standard library
-		}
-		var items []listItem
-		dec := json.NewDecoder(&stdout)
-		for {
-			var item listItem
-			if err := dec.Decode(&item); err == io.EOF {
-				break
-			} else if err != nil {
-				return nil, fmt.Errorf("parsing `go list` output: %w", err)
-			}
-			items = append(items, item)
-		}
-
-		output := make(map[string]string, len(items))
-		var err error
-		for _, item := range items {
-			if item.Standard && item.ImportPath == "unsafe" && item.Export == "" {
-				// Special-casing "unsafe", because it's not provided like other modules
-				continue
-			}
-			if item.Export == "" {
-				err = errors.Join(err, fmt.Errorf("%s (%s) has no export file", item.ImportPath, item.BuildID))
-				continue
-			}
-			output[item.ImportPath] = item.Export
-		}
+	cmd := exec.Command("go", args...)
+	var stdout bytes.Buffer
+	cmd.Stdout = &stdout
+	log.Tracef("Attempting to resolve %q using %q\n", importPath, cmd.Args)
+	if err = cmd.Run(); err != nil {
+		return nil, fmt.Errorf("running %q: %w", cmd.Args, err)
+	}
 
-		if err == nil {
-			return output, nil
-		} else if attemptedBuild {
-			return nil, fmt.Errorf("after `go build`: %s", err)
+	type listItem struct {
+		ImportPath string // The import path of the package
+		Export     string // The path to its archive, if any
+		BuildID    string // The build ID for the package
+		Standard   bool   // Whether this is from the standard library
+	}
+	var items []listItem
+	dec := json.NewDecoder(&stdout)
+	for {
+		var item listItem
+		if err := dec.Decode(&item); err == io.EOF {
+			break
+		} else if err != nil {
+			return nil, fmt.Errorf("parsing `go list` output: %w", err)
 		}
```

Comment:
No `else` plz.
```suggestion
		}
		if err != nil {
			return nil, fmt.Errorf("parsing `go list` output: %w", err)
		}
```

Reply:
The original `if` branch of that `else` shadows some `err` with its own, and removing the `else` actually changes the meaning of this condition (👏🏻 thank you go, you never disappoint 👏🏻 )

Reply:
I would have liked for you to still remove it because the fact that you had had to explain me that makes this code even more confusing. But since this is the only change you would have to make, I will not die on this hill